src/components/common/filtering/useProject.tsx

```diff
@@ -0,0 +1,145 @@
+import { useCallback, useEffect, useState } from "react";
+import agent from "../../../utils/agent";
+import { Project, ProjectLite, ProjectParams } from "./project";
+
+export default function useProjects() {
+  const [projects, setProjects] = useState<ProjectLite[]>([]);
+  const [projectsLoaded, setProjectsLoaded] = useState(false);
+  const [projectsLoading, setProjectsLoading] = useState(false);
+  const [filteredProjects, setFilteredProjects] = useState<ProjectLite[]>([]);
+  const [positions, setPositions] = useState<string[]>([]);
+  const [positionsLoaded, setPositionsLoaded] = useState(false);
+  const [commitments, setCommitments] = useState<string[]>([]);
+  const [commitmentsLoaded, setCommitmentsLoaded] = useState(false);
+
+  const SetProjectParams = (value: ProjectParams) => {
+    setFilteredProjects(FilterProjects(projects, value));
+  };
+
+  /**
+   * Request the projects from the API
+   */
+  const fetchProjects = useCallback(async () => {
+    try {
+      const result: Project[] = await agent.Projects.list();
+      if (result.length > 0) {
+        const list = result.map((item: Project) => ({
+          id: item.id,
+          slug: item.slug,
+          catchPhrase: item.catchPhrase,
+          title: item.title,
+          description: item.description,
+          commitmentLevel: item.commitmentLevel,
+          openPositions: item.openPositions,
+        }));
+        setProjects(list);
+        setFilteredProjects(list);
+        setProjectsLoaded(true);
+        fetchPositions(result);
+        fetchCommitments(result);
+      }
+    } catch (error) {
+      console.log(error);
+    } finally {
+      setProjectsLoading(false);
+    }
+  }, []);
+
+  /**
+   * Extract the Open Positions from the projects entities
+   */
+  function fetchPositions(projects: Project[]) {
+    if (projects.length <= 0) return;
+    const items = projects
+      .flatMap((p) => p.openPositions.map((po) => po.title))
+      .sort();
+    const distinct = Array.from(new Set(items));
+    setPositions(distinct);
+    setPositionsLoaded(true);
+  }
```

Comment:
we need to add an endpoint in the api for fetching Positions rather than doing it this way. maybe "/positions"